app/experimenter/experiments/views.py

```diff
@@ -327,6 +336,99 @@ def get_initial(self):
         return initial
 
 
+def clone_experiment_name(name):
+    version = re.search(r"v(\d+)$", name)
+    if version:
+        new_version = name.replace(
+            version.group(1), str(int(version.group(1)) + 1)
+        )
+    else:
+        new_version = name + " v2"
+
+    if Experiment.objects.filter(name=new_version):
+        return clone_experiment_name(new_version)
+
+    return new_version
+
+
+class ExperimentCloneView(ExperimentFormMixin, View):
+
+    def post(self, request, *args, **kwargs):
+        experiment = Experiment.objects.filter(slug=self.kwargs["slug"]).get()
+        old_experiment_name = experiment.name
+        variants = ExperimentVariant.objects.filter(experiment=experiment)
+
+        set_to_none_fields = [
+            "addon_experiment_id",
+            "addon_release_url",
+            "normandy_slug",
+            "normandy_id",
+            "review_science",
+            "review_engineering",
+            "review_qa_requested",
+            "review_intent_to_ship",
+            "review_bugzilla",
+            "review_qa",
+            "review_relman",
+            "review_advisory",
+            "review_legal",
+            "review_ux",
+            "review_security",
+            "review_vp",
+            "review_data_steward",
+            "review_comms",
+            "review_impacted_teams",
+        ]
+
+        # duplicate the old experiment by setting id=none, clearing
+        # appropriate fields, then calling save()
+
+        experiment.id = None
+        experiment.name = clone_experiment_name(experiment.name)
+        experiment.slug = slugify(experiment.name)
+        experiment.status = ExperimentConstants.STATUS_DRAFT
+
+        experiment.owner = get_user_model().objects.get(
+            id=self.request.user.id
+        )
+        experiment.parent = Experiment.objects.filter(
+            slug=self.kwargs["slug"]
+        ).get()
+        experiment.proposed_start_date = datetime.date.today()
+        experiment.archived = False
```

Comment:
Yep good idea.  Actually this is kind of fun, you can create an archived template and then clone it a bunch.